numpy/ma/tests/test_core.py

```diff
@@ -5387,7 +5387,7 @@ def test_ufunc_with_out_varied():
 
 
 def test_astype_mask_ordering():
-    descr = [('v', int, 3), ('x', [('y', float)])]
+    descr = np.dtype([('v', int, 3), ('x', [('y', float)])])
```

Comment:
So record dtypes are now nominal not structural? That is, two record types with identical fields are treated differently depending on whether they are at the same memory locations? That seems undesirable to me; although if structured dtypes have mutable state maybe it's the right choice after all.

Either way, it would be good to test this explicitly rather than only accidentally as part of this test

Reply:
Well, we can either preserve the array identity check faithfully, or the dtype identity check one, since doing `np.dtype(something)` will give a new dtype each time.

Mutibility is a point (although I don't like it for dtypes).  I guess the structured dtype case is "more equivalent" than for the ones with differing type numbers...
But is there much point in preserving the array identity?

Reply:
I renamed the new test function to make it (hopefully) more obvious that it includes tests of the new dtype equivalence/identity behavior. Does those assertions seem sufficient?

Reply:
I think the latest test is pretty thorough and illustrative. If a different test file name would help to find the expression of these identity semantics, please advise.